plutus-tx-plugin/test/Map/Spec.hs

```diff
@@ -0,0 +1,94 @@
+-- editorconfig-checker-disable-file
+{-# LANGUAGE BangPatterns          #-}
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE LambdaCase            #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NegativeLiterals      #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE PatternSynonyms       #-}
+{-# LANGUAGE TemplateHaskell       #-}
+{-# LANGUAGE ViewPatterns          #-}
+{-# OPTIONS_GHC -fplugin-opt PlutusTx.Plugin:context-level=0 #-}
+{-# OPTIONS_GHC -fplugin-opt PlutusTx.Plugin:defer-errors #-}
+
+module Map.Spec where
+
+import Test.Tasty.Extras
+
+import PlutusTx.Builtins qualified as PlutusTx
+import PlutusTx.Code
+import PlutusTx.DataMap qualified as Map
+import PlutusTx.Lift (liftCodeDef)
+import PlutusTx.Prelude qualified as PlutusTx
+import PlutusTx.Show qualified as PlutusTx
+import PlutusTx.Test
+import PlutusTx.TH (compile)
+
+tests :: TestNested
+tests =
+  testNestedGhc
+    "Budget"
+    [ goldenPirReadable "map1" map1
+    , goldenUPlcReadable "map1" map1
+    , goldenEvalCekCatch "map1" $ [map1 `unsafeApplyCode` (liftCodeDef 100)]
+    , goldenBudget "map1-budget" $ map1 `unsafeApplyCode` (liftCodeDef 100)
+    , goldenPirReadable "map2" map2
+    , goldenUPlcReadable "map2" map2
+    , goldenEvalCekCatch "map2" $ [map2 `unsafeApplyCode` (liftCodeDef 100)]
+    , goldenBudget "map2-budget" $ map2 `unsafeApplyCode` (liftCodeDef 100)
+    ]
+
+map1 ::
+  CompiledCode
+    ( Integer ->
+      ( Maybe PlutusTx.BuiltinByteString
+      , Maybe PlutusTx.BuiltinByteString
+      , Maybe PlutusTx.BuiltinByteString
+      , Maybe PlutusTx.BuiltinByteString
+      , Maybe PlutusTx.BuiltinByteString
+      )
+    )
+map1 =
```

Comment:
A short description of what we're testing here?

Reply:
Also I think it would be appropriate to do some "model" property-based tests where we test that this map behaves like a Haskell map and/or AssocMap under various series of operations.